tests/ui/feature-gates/feature-gate-unsafe_fields.rs

```diff
@@ -0,0 +1,23 @@
+//@ compile-flags: --crate-type=lib
+//@ revisions: with_gate without_gate
+//@ [with_gate] check-pass
+
+#![allow(incomplete_features)]
+#![cfg_attr(with_gate, feature(unsafe_fields))]
+
+struct Foo {
+    unsafe field: (), //[without_gate]~ ERROR
+}
+
+// This should not parse as an unsafe field definition.
+struct FooTuple(unsafe fn());
```

Comment:
```suggestion
#[cfg(any())]
struct FooTuple(unsafe fn());
```

Reply:
An error should be emitted in cfg'd-out code without the gate. There have been past instances of syntax accidentally stabilized in cfg'd-out code, so it's best to test this just t obe safe.

Reply:
This syntax is not an error in today's Rust ;) (it's declaring a field that is an unsafe fn pointer)